staging/src/k8s.io/api/resource/v1beta2/types.go

```diff
@@ -804,7 +804,9 @@ type DeviceSubRequest struct {
 	// +optional
 	AllocationMode DeviceAllocationMode `json:"allocationMode,omitempty" protobuf:"bytes,4,opt,name=allocationMode"`
 
-	// Count is used only when the count mode is "ExactCount". Must be greater than zero.
+	// Count is used only when the count mode is "ExactCount". Must be greater than zero
+	// unless the subrequest is the last entry in the FirstAvailable list. For the last entry,
+	// a count of zero is allowed.
 	// If AllocationMode is ExactCount and this field is not specified, the default is one.
```

Comment:
Unless it's the last entry? This isn't a pointer type, so I don't think you can tell if it's unspecified or zero?

Reply:
what was our reasoning behind using `ExactCount: 0` vs a new allocation mode, e.g. `AllocationMode: None`?

Reply:
This was discussed a bit in https://github.com/kubernetes/kubernetes/issues/131194, but essentially we thought that the use-case for setting `ExactCount: 0` to be somewhat narrow so adding a new enum for that didn't seem necessary.

But I think the comment from @tallclair here refers to the inaccurate comment on line 810, which seems to suggest that the defaulting logic only applies if the `Count` field is not specified. This is not true, since we can distinguish between the field set to 0 and not being set at all. So our defaulting logic kicks in if the field is set to 0. This is not really specific to this PR, but I've updated the comment to be more accurate.